### DISCUSSION TITLE
How to change the default ADC pin(GPIO34)

### DISCUSSION THREAD

**Discussion by lgherca** [pschatzmann/arduino-audio-tools#783] [Q&A] [answered]
Hello everyone! I am trying to do a 32 band spectrum analyzer using FFT and AudioTools but I cannot use a different analog input other than the default GPIO34.
The below code only works if AUDIO_IN_PIN is defined as 34. I have tried values of 32 and 36 based on my HW availability. What am I missing?

#include "Arduino.h"
#include "fft.h"
#include "AudioTools.h"
...
#define SAMPLING_FREQ   40000 
#define AUDIO_IN_PIN    34

AnalogAudioStream adc;
...
void setup() {
  // put your setup code here, to run once:
  // put your setup code here, to run once:
  Serial.begin(115200);
  Serial.println("Booting"); 

//Setup FFT
  real_fft_plan = fft_init(SAMPLES, FFT_REAL, FFT_FORWARD, NULL, NULL);
  
//Setup ADC
  auto cfgRx = adc.defaultConfig(RX_MODE);
  cfgRx.sample_rate = SAMPLING_FREQ;
  cfgRx.channels = 1;
  cfgRx.setInputPin1(AUDIO_IN_PIN); 
  adc.begin(cfgRx);

//Setup FastLED
  FastLED.addLeds<CHIPSET, LED_PIN, COLOR_ORDER>(leds, NUM_LEDS).setCorrection(TypicalSMD5050);
  FastLED.setMaxPowerInVoltsAndMilliamps(LED_VOLTS, MAX_MILLIAMPS);
  FastLED.setBrightness(BRIGHTNESS_SETTINGS[1]);
  FastLED.clear();
}

**Comment by pschatzmann** [comment, marked as answer]
I would expect that all pins from ADC1 should work.

**Reply by lgherca** [reply]
GPIO32 and GPIO36 belong to ADC1, so based on your expectation it should work. However, on my side it doesn't work. What could be the issue? Is this setup code ok?
auto cfgRx = adc.defaultConfig(RX_MODE);
cfgRx.sample_rate = SAMPLING_FREQ;
cfgRx.channels = 1;
cfgRx.setInputPin1(AUDIO_IN_PIN);
adc.begin(cfgRx);

**Comment by pschatzmann** [comment]
yes, this looks right to me...
I don't see what could be wrong.

**Reply by lgherca** [reply]
I tried other pins and in the end GPIO33 also worked(so at least I have 2 pins working). Probably there is an issue with my hardware. Thanks!
